src/Microsoft.DotNet.ProjectModel/Files/IncludeContext.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.ProjectModel.Files
+{
+    public class IncludeContext
+    {
+        private static readonly char[] PatternSeparator = new[] { ';' };
+
+        public IncludeContext(
+            string sourceBasePath,
+            string option,
+            JObject rawObject,
+            string[] defaultBuiltInInclude = null,
+            string[] defaultBuiltInExclude = null)
+        {
+            SourceBasePath = sourceBasePath;
+            Option = option;
+            var token = rawObject.Value<JToken>(option);
+            if (token.Type != JTokenType.Object)
+            {
+                IncludePatterns = new List<string>(ExtractValues(token));
+            }
+            else
+            {
+                IncludePatterns = CreateCollection(sourceBasePath, "include", ExtractValues(token.Value<JToken>("include")), literalPath: false);
+                ExcludePatterns = CreateCollection(sourceBasePath, "exclude", ExtractValues(token.Value<JToken>("exclude")), literalPath: false);
+                IncludeFiles = CreateCollection(sourceBasePath, "includeFiles", ExtractValues(token.Value<JToken>("includeFiles")), literalPath: true);
+                ExcludeFiles = CreateCollection(sourceBasePath, "excludeFiles", ExtractValues(token.Value<JToken>("excludeFiles")), literalPath: true);
+                var builtIns = token.Value<JToken>("builtIns") as JObject;
+                if (builtIns != null)
+                {
+                    BuiltInsInclude = CreateCollection(sourceBasePath, "include", ExtractValues(builtIns.Value<JToken>("include")), literalPath: false);
+                    if (defaultBuiltInInclude != null && !BuiltInsInclude.Any())
+                    {
+                        BuiltInsInclude = defaultBuiltInInclude.ToList();
+                    }
+                    BuiltInsExclude = CreateCollection(sourceBasePath, "exclude", ExtractValues(builtIns.Value<JToken>("exclude")), literalPath: false);
+                    if (defaultBuiltInExclude != null && !BuiltInsExclude.Any())
+                    {
+                        BuiltInsExclude = defaultBuiltInExclude.ToList();
+                    }
+                }
+                var mappings = token.Value<JToken>("mappings") as JObject;
+                if (mappings != null)
+                {
+                    Mappings = new List<KeyValuePair<string, IncludeContext>>();
+                    foreach (var map in mappings)
+                    {
+                        Mappings.Add(new KeyValuePair<string, IncludeContext>(map.Key, new IncludeContext(sourceBasePath, map.Key, mappings)));
+                    }
+                }
```

Comment:
please add more empty lines between structures, this piece is not very readable.